pkg/apis/cluster/v1alpha1/machine_types.go

```diff
@@ -70,7 +70,7 @@ type MachineSpec struct {
 	// should populate the values it uses when persisting Machine objects.
 	// A Machine spec missing this field at runtime is invalid.
 	// +optional
-	Versions MachineVersionInfo `json:"versions,omitempty"`
+	Versions *MachineVersionInfo `json:"versions,omitempty"`
```

Comment:
All consumers of the field will be required to change the field access logic. Their code will start to panic in cases where the `Versions` field is nil.